src/worker/ghidra/ghidra.cc

```diff
@@ -0,0 +1,141 @@
+#include <loadimage.hh>
+
+#include <iostream>
+#include <exception>
+#include <typeinfo>
+#include <stdexcept>
+
+#include "sleighMishegos.hh"
+#include "../worker.h"
+#include "mish_common.h"
+
+extern "C" {
+char *worker_name = (char *)"ghidra";
+
+void worker_ctor();
+
+void try_decode(decode_result *result, uint8_t *raw_insn, uint8_t length);
+}
+
+// This is a tiny LoadImage class which feeds the executable bytes to the translator
+class MyLoadImage : public LoadImage {
+  uintb baseaddr;
+  int4 length;
+  uint1 *data;
+
+public:
+  MyLoadImage(uintb ad, uint1 *ptr, int4 sz) : LoadImage("nofile") {
+    baseaddr = ad;
+    data = ptr;
+    length = sz;
+  }
+  virtual void loadFill(uint1 *ptr, int4 size, const Address &addr);
+  virtual string getArchType(void) const {
+    return "x86:LE:64:default";
+  }
+  virtual void adjustVma(long adjust) {
+  }
+  virtual void setData(uint1 *ptr, int4 sz) {
+    this->data = ptr;
+    this->length = sz;
+  }
+};
+
+// This is the only important method for the LoadImage. It returns bytes from the static array
+// depending on the address range requested
+void MyLoadImage::loadFill(uint1 *ptr, int4 size, const Address &addr) {
+  uintb start = addr.getOffset();
+  uintb max = baseaddr + (length - 1);
+  for (int4 i = 0; i < size; ++i) {              // For every byte requestes
+    uintb curoff = start + i;                    // Calculate offset of byte
+    if ((curoff < baseaddr) || (curoff > max)) { // If byte does not fall in window
+      ptr[i] = 0;                                // return 0
+      continue;
+    }
+    uintb diff = curoff - baseaddr;
+    ptr[i] = data[(int4)diff]; // Otherwise return data from our window
+  }
+}
+
+// Here is a simple class for emitting assembly.  In this case, we send the strings straight
+// to the result.
+class AssemblyMishegos : public AssemblyEmit {
+  decode_result *result;
+
+public:
+  AssemblyMishegos(decode_result *dr) : result(dr){};
+  virtual void dump(const Address &addr, const string &mnem, const string &body) {
+    // std::cout << ": " << mnem << ' ' << body << std::endl;
+    result->status = S_SUCCESS;
+    result->len =
+        snprintf(result->result, MISHEGOS_DEC_MAXLEN, "%s %s\n", mnem.c_str(), body.c_str());
+  }
+};
+
+static const uintb START_ADDRESS = 0x0;
```

Comment:
Does starting address matter at all for disassembly listings? I think others use address 0, so hopefully that's all normalized.